Antidote/CallsManager.m

```diff
@@ -451,13 +465,33 @@ - (void)switchViewControllerForCall:(OCTCall *)call
 
     self.currentCallViewController = abstractVC;
 
-    self.currentCall = call;
+    self.currentCall = [RBQSafeRealmObject safeObjectFromObject:call];
 }
 
-- (BOOL)onlyPauseCallsLeft
+- (void)replaceCurrentCallThatWasRemoved
 {
-    return (([self.allActiveCallsController.fetchedObjects count] == 0) &&
-            ([self.allPausedCallsController.fetchedObjects count] != 0));
+    OCTCall *call = [self.allPausedCallsController.fetchedObjects firstObject];
+
+    if (! call) {
+        call = [self.allCallsController.fetchedObjects firstObject];
+    }
+
+    // workaround for deadlock in objcTox
+    // https://github.com/Antidote-for-Tox/objcTox/issues/51
+    [self performSelector:@selector(switchViewControllerForCall:) withObject:call afterDelay:0];
```

Comment:
If there are no calls from the pause calls, we want to grab a call from the others, which would be incoming calls. 

For example, Alice is in active call with Bob, Carol makes an incoming call. Alice ends call with Bob. In this case, there are no paused calls, so this method will take the incoming one.